src/ie9-ajax.js

```diff
@@ -51,15 +53,24 @@ function ie9Ajax(method, url, opts) {
       }
     };
 
-    function handleError() {
-      reject(new Error('There was an error with the XDR'));
+    function handleError(error) {
+      logger.error(`There was an error with the XDR\n${JSON.stringify(error, null, '  ')}`);
     }
 
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        logger.error('Please ensure Shopify JavaScript Buy SDK is run over https');
+      } else {
+        logger.error(`There was an error when running XDR\n${error.message}`);
```

Comment:
so IE (I think only < 10) does this thing where the first time you ever encounter a js error on any webpage it will popup a window telling you about it, then if you _uncheck_ the checkbox in that dialog it will not show you those popups again. Many people do not uncheck this checkbox. If you click the "Show Details" button, that's what usually crashes the browser in my experience. 

I think the defaults for this behaviour changed several times during the lifetime of IE9